linode/lke/cluster.go

```diff
@@ -363,7 +375,7 @@ func matchPoolsWithSchema(pools []linodego.LKENodePool, declaredPools []interfac
 		declaredPool := declaredPool.(map[string]interface{})
 
 		for key, pool := range poolMap {
-			if pool.Count != declaredPool["count"] || pool.Type != declaredPool["type"] {
+			if pool.ID != declaredPool["id"] {
```

Comment:
Turns out we can just select on ID because `declaredPool` is pulled from state 🙂